### PR TITLE
Push the mark before going to definition (using etags)

### PR DESCRIPTION
**Pull request by delexi**
See discussion for #992.

Going to definition is usually a "bigger" jump - which makes it feasable
to push the mark. This allows to go back after looking up a definition
of a symbol.

- Add fsharp-ac/pop-gotodefn-stack which pops the goto definition stack
  and thus goes back to where point was before jumping to definition.
- This uses etags.
- Add key bindings M-. and M-, and C-c C-b to mode map and to menu.
- Extend integration test for goto definition.
- Readme: convert key bindings list to table and add new key bindings.

### CODE REVIEW

File: emacs/README.md [hunk 1]
```diff
@@ -148,25 +148,27 @@ There are a few variables you can adjust to change how fsharp-mode behaves:
...
-- <kbd>C-c l</kbd>:         Shift region to left
-- <kbd>C-c r</kbd>:         Shift region to right
-- <kbd>C-c &lt;up&gt;</kbd>:      Move cursor to the beginning of the block
-- <kbd>C-c C-p</kbd>:       Load a project for autocompletion and tooltips
-- <kbd>C-c C-d</kbd>:       Jump to definition of symbol at point
-- <kbd>C-c C-t</kbd>:       Request a tooltip for symbol at point
-- <kbd>C-c C-q</kbd>:       Quit current background compiler process
-- <kbd>M-n</kbd>:           Go to next error
-- <kbd>M-p</kbd>:           Go to previous error
+ Key binding                         | Description
```
Comment (rneatherway) [inline emacs/README.md]: This table doesn't render properly in the Github markdown view: https://github.com/delexi/fsharpbinding/blob/push-mark-before-goto-defn-2/emacs/README.md

**Comment by rneatherway** [comment]
It looks good apart from the table. The builds are not your fault, I need to check what is going on there.

File: emacs/README.md (hunk 1 above)
Comment (delexi) [inline emacs/README.md]: I missed a blank line before the table.

**Comment by delexi** [comment]
I amended my old commit with the fixed table.

File: emacs/fsharp-mode-completion.el [hunk 2]
```diff
@@ -714,6 +720,7 @@ around to the start of the buffer."
          (line (gethash "Line" data))
          (col (gethash "Column" data)))
     (find-file file)
+    (ring-insert find-tag-marker-ring (point-marker))
```
Comment (rneatherway) [inline emacs/fsharp-mode-completion.el]: This needs to be before `find-file`

**Comment by rneatherway** [comment]
Things work with that change, but for some reason the visit definition call in the background doesn't affect the current buffer when in test mode. I don't understand this behaviour, but this is why your extended test was failing. If you make that change I will merge. Maybe I'll ask on stack exchange if anyone can see why changing files doesn't work.

**Comment by rneatherway** [comment]
By the way, `/=` doesn't work for strings (just numbers and markers) in case that comes up again. You should use `equal` in that case.

File: emacs/fsharp-mode-completion.el (hunk 2 above)
Comment (delexi) [inline emacs/fsharp-mode-completion.el]: Of course!

**Comment by delexi** [comment]
I moved the line up and amended the commit!

**Comment by rneatherway** [comment]
Thanks for this really handy contribution!

**Pull request merged**